packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/refetch-data.jsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import {useHistory, useLocation} from 'react-router-dom'
+import {useQueryClient} from '@tanstack/react-query'
+import LoadingSpinner from '../loading-spinner'
+
+// For good UX, show loading spinner long enough for users to see
+const LOADING_SPINNER_MIN_DURATION = 500
+
+const RefetchData = () => {
+    const history = useHistory()
+    const location = useLocation()
+
+    let queryClient
+    try {
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        queryClient = useQueryClient()
+    } catch (err) {
+        // `useQueryClient` throws an error if the project does not use react-query.
+        // So in this case, leave `queryClient` as undefined. Continue to navigate to the referrer.
+    }
+
+    useEffect(() => {
+        const refetchData = async () => {
+            const showLoadingSpinner = new Promise((resolve) =>
+                setTimeout(resolve, LOADING_SPINNER_MIN_DURATION)
+            )
+            const invalidateQueries = queryClient
+                ? queryClient.invalidateQueries()
+                : Promise.resolve()
```

Comment:
```suggestion
            const invalidateQueries = queryClient?.invalidateQueries()
```
Totally inconsequential, but `Promise.all` is perfectly happy to accept `undefined` in the array. 🙂